src/refs.c

```diff
@@ -91,18 +91,23 @@ git_reference *git_reference__alloc(
 	return ref;
 }
 
-git_reference *git_reference__set_name(
-	git_reference *ref, const char *name)
+git_reference *git_reference__realloc(
+	git_reference **ptr_to_ref, const char *name)
```

Comment:
Wouldn't it make more sense to have it return an error code and instead let it update the pointer in-place? Feel free to leave it as-is though.

Reply:
Not sure. I wanted to make the call "usable, but in a safe way". If it's implementation-defined whether realloc will realloc to a smaller size, and I need to know if the old ref is now invalid, having the pointer be NULL on return signals that case, and the higher-level code can adapt.

Arguably, most of that code comes in preparation of some more work 😉, but I want to keep it public API/documentation specific there. For example, I think having a "higher" low watermark on refname lengths could help with the invalidation (something like 32 could skip quite a few reallocs).

Reply:
Yeah, just leave it as-is then. Your change is an improvement compared to the current code and doesn't introduce anything new that one could complain about, so it can only be a win for us.